tide_media.install

```diff
@@ -243,3 +244,81 @@ function tide_media_update_8005() {
     }
   }
 }
+
+/**
+ * Video transcript field change from text field to WYSIWYG.
+ */
+function tide_media_update_8006() {
+  $entity_type = 'media';
+  $field_name = 'field_media_transcript';
+  $field_storage = FieldStorageConfig::loadByName($entity_type, $field_name);
+
+  if (is_null($field_storage)) {
+    return;
+  }
+
+  $database = Drupal::database();
+  $line_breaks = [
+    "\r\n",
+    "\n",
+    "\r",
+  ];
+  $br = "<br />";
+  // Step1 create two temporary tables for holding data.
+  $tmp_table_names = [
+    'tmp_table' => 'media__field_media_transcript',
+    'tmp_revision_table' => 'media_revision__field_media_transcript',
+  ];
+  $spec = [
+    'type' => 'text',
+    'size' => 'big',
+    'not null' => FALSE,
+  ];
+
+  foreach ($tmp_table_names as $tmp => $table_name) {
+    if ($database->schema()->tableExists($table_name)) {
+      $database->query("CREATE TABLE {$tmp} SELECT * FROM  {$table_name}");
+      $database->truncate($table_name)->execute();
+      $database->schema()->addField($table_name, 'field_media_transcript_format', $spec);
+    }
+  }
+
+  // Step2 update field config from string_long to text_long.
+  // Update FieldStorageConfig.
+  $new_field_storage = $field_storage->toArray();
+  $new_field_storage['type'] = 'text_long';
+  $new_field_storage['module'] = 'text';
+  $new_field_storage = FieldStorageConfig::create($new_field_storage);
+  $new_field_storage->original = $new_field_storage;
+  $new_field_storage->enforceIsNew(FALSE);
+  $new_field_storage->save();
+
+  // Update field.
+  foreach ($field_storage->getBundles() as $bundle => $label) {
+    $field = FieldConfig::loadByName($entity_type, $bundle, $field_name);
+    $new_field = $field->toArray();
+    $new_field['field_type'] = 'text_long';
+    $new_field = FieldConfig::create($new_field);
+    $new_field->original = $field;
+    $new_field->enforceIsNew(FALSE);
+    $new_field->save();
+  }
+
+  // Step 3 data migration.
+  foreach ($tmp_table_names as $tmp => $table_name) {
+    $database->schema()->addField($tmp, 'field_media_transcript_format', $spec);
+    $database->update($tmp)
+      ->fields(['field_media_transcript_format' => 'rich_text'])
+      ->execute();
+    $database->insert($table_name)->from($database->select($tmp, 't')
+      ->fields('t'))->execute();
+    foreach ($line_breaks as $line_break) {
+      $database->update('media__field_media_transcript')->expression('field_media_transcript_value', "REPLACE(field_media_transcript_value,:line_break,:br)", [
```

Comment:
@vincent-gao do we also need to update the `media_revision__field_media_transcript` table?

Reply:
@sonnykt yes. typo issue. thanks Sonny.